wgengine/netstack/netstack.go

```diff
@@ -1457,16 +1485,23 @@ func (ns *Impl) acceptUDP(r *udp.ForwarderRequest) {
 		return
 	}
 
-	// Handle magicDNS traffic (via UDP) here.
+	// Handle magicDNS and loopback traffic (via UDP) here.
 	if dst := dstAddr.Addr(); dst == serviceIP || dst == serviceIPv6 {
-		if dstAddr.Port() != 53 {
+		switch {
+		case dstAddr.Port() != 53 && !ns.isLoopbackPort(dstAddr.Port()):
 			ep.Close()
-			return // Only MagicDNS traffic runs on the service IPs for now.
+			return // Only MagicDNS and loopback traffic runs on the service IPs for now.
+		case ns.isLoopbackPort(dstAddr.Port()):
+			if dst == serviceIPv6 {
+				dstAddr = netip.AddrPortFrom(ipv6Loopback, dstAddr.Port())
+			} else {
+				dstAddr = netip.AddrPortFrom(ipv4Loopback, dstAddr.Port())
+			}
+		default:
```

Comment:
it's pretty tempting to flip this around now to dstAddr.Port() == 53 now, rather than this way around, we certainly should if this grows anymore

Reply:
addressed in https://github.com/tailscale/tailscale/pull/13301/commits/da6d77d5ea55d7efd6d08f883f0febdfb3312650